src/app/services/todos-store.service.ts

```diff
@@ -103,32 +103,54 @@ export class TodosStoreService {
     const todo = this.todos.find((t) => t.id === id);
     this.todos = this.todos.filter((t) => t.id !== id);
 
-    // TODO: Undo operation
-    // Timeout server remove by notifier duration
-    // And then reset using notifier's action
-
-    if (serverRemove) {
-      this.todosService.remove(id).subscribe(
-        () => {},
-        (error) => {
-          this.notifier.open(`Error: ${error.message}`);
-          setTimeout(() => (this.todos = [...this.todos, todo]));
-        }
-      );
-    }
+    // Prompt user if undo is necessary
+    const mode = { undo: false };
+    const msDuration = 3000;
+    const revertCb = () => {
+      mode.undo = true;
+      this.todos = [...this.todos, todo];
+    };
+    this.notifier.undoAlert(mode, revertCb, `Removed Todo.`, msDuration);
```

Comment:
it is better to call `undoAlert` after a task is successfully deleted and then re-create the deleted task on the server if a user clicks on the undo buttons